planner/core/exhaust_physical_plans.go

```diff
@@ -1744,9 +1747,15 @@ func (p *LogicalJoin) tryToGetMppHashJoin(prop *property.PhysicalProperty, useBC
 		return nil
 	}
 
-	if (p.JoinType != InnerJoin && p.JoinType != LeftOuterJoin && p.JoinType != RightOuterJoin && p.JoinType != SemiJoin && p.JoinType != AntiSemiJoin) || len(p.EqualConditions) == 0 {
+	if p.JoinType != InnerJoin && p.JoinType != LeftOuterJoin && p.JoinType != RightOuterJoin && p.JoinType != SemiJoin && p.JoinType != AntiSemiJoin {
 		return nil
 	}
+
+	if len(p.EqualConditions) == 0 {
+		if p.ctx.GetSessionVars().AllowCARTESIANBCJ < 1 || !useBCJ {
```

Comment:
Should this be `< 1` or  `< 2`?

Reply:
`= 1` means `allow carstesian broadcast join`, `= 2` means `force carstesian broadcast join`, so should use `< 1` here.